### DISCUSSION TITLE
Error: no+cuda+implementation+for+layer-norm

### DISCUSSION THREAD

**Discussion by siddthartha** [huggingface/candle#2217] [Q&A] [answered]
Trying to build stable-diffusion pipeline from examples I've got it when set cpu parameter to false.. Am I do something wrong or there is no enough implementation for SD for now?

**Comment by LaurentMazare** [comment]
That might be due to some changes made yesterday. If you're running the example within the repo can you provide the exact command you're using? If it's with a custom Cargo.toml file, ensure that the cuda feature is enabled on all of candle-core, candle-nn, and candle-transformers.

**Reply by siddthartha** [reply]
I moved it from examples to my worker, and made Cargo.toml like this
```
...
candle-core = { git = "https://github.com/huggingface/candle.git", version = "0.5.1", features = ["cuda"] }
candle-examples = { git = "https://github.com/huggingface/candle.git", version = "0.5.1" }
candle-transformers = { git = "https://github.com/huggingface/candle.git", version = "0.5.1" }
candle-nn = { git = "https://github.com/huggingface/candle.git", version = "0.5.1" }
hf-hub = "0.3.2"
tokenizers = "0.19.1"
...
```
Version for cpu works (after commenting set_seed() becourse of `cannot seed the CPU rng with set_seed`). Then i've got a generated image. But after switching to cuda (add features=["cuda"]) -- got this one error with layer_norm. Code of pipeline is the same as in example (it's here https://github.com/siddthartha/enso-ml/blob/d1f17936c04d4c437609086ee2dcc0730c25c742/src/lib.rs#L119). I run biuld just like `cargo build --release && ./target` on a cloud side (with GPU).

**Reply by LaurentMazare** [reply, marked as answer]
I think you also need to enable the `cuda` feature on `candle-nn`.

**Reply by siddthartha** [reply]
Yupp, it works. Thanks a lot!
